Authenticator/Source/Root.swift

```diff
@@ -66,15 +68,36 @@ class AppModel {
             modal = .EditForm(form.viewModel)
         }
 
-        return AppViewModel(
+        return RootViewModel(
             tokenList: tokenList.viewModel,
             modal: modal
         )
     }
 }
 
-extension AppModel {
-    func handleAction(action: AppAction) {
+extension Root {
+    enum Action {
+        case BeginTokenEntry
+        case BeginManualTokenEntry
+        case CancelTokenEntry
+        case SaveNewToken(Token)
+
+        case BeginTokenEdit(PersistentToken)
+        case CancelTokenEdit
+        case SaveChanges(Token, PersistentToken)
+
+        case AddTokenFromURL(Token)
+
+        case TokenListAction(TokenList.Action)
+        case TokenEntryFormAction(TokenEntryForm.Action)
+        case TokenEditFormAction(TokenEditForm.Action)
+
+        case UpdateToken(PersistentToken)
+        case MoveToken(fromIndex: Int, toIndex: Int)
+        case DeletePersistentToken(PersistentToken)
+    }
+
+    func handleAction(action: Action) {
```

Comment:
Function Body Length Violation: Function body should span 40 lines or less: currently spans 78 lines (function_body_length)